app/src/main/java/com/gathering/android/NavHostActivity.kt

```diff
@@ -2,18 +2,39 @@ package com.gathering.android
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import androidx.navigation.ui.setupWithNavController
 import com.gathering.android.databinding.ActNaveHostBinding
+import com.gathering.android.signup.model.AuthRepository
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class NavHostActivity : AppCompatActivity() {
 
     private lateinit var binding: ActNaveHostBinding
 
+    @Inject
+    lateinit var authRepository: AuthRepository
```

Comment:
You should not have a repository in the view layer directly, you need to have a VM here.

Reply:
It was solved. I created the ViewModel for that